### PR TITLE
Fix broken FlushAll

### PR DESCRIPTION
**Pull request by zeripath**
Go contexting is not quite how you expect...

```go
a := ["1", "2", "3"]
for _, i := range a {
   go func() {
      fmt.Println(i)
    } ()
}
```
will most likely print 3 3 3.

Our linter should have caught this as go vet would catch it

### CODE REVIEW

File: modules/queue/manager.go [hunk 1]
```diff
@@ -183,14 +183,15 @@ func (m *Manager) FlushAll(baseCtx context.Context, timeout time.Duration) error
 			}
 			allEmpty = false
 			if flushable, ok := mq.Managed.(Flushable); ok {
+				ourmq := mq
 				go func() {
```
Comment (lafriks) [inline modules/queue/manager.go]:
```suggestion
				go func(q *ManagedQueue) {
```

File: modules/queue/manager.go [hunk 2]
```diff
@@ -183,14 +183,15 @@ func (m *Manager) FlushAll(baseCtx context.Context, timeout time.Duration) error
...
+					pid := ourmq.RegisterWorkers(1, start, hasTimeout, end, localCancel, true)
 					err := flushable.FlushWithContext(localCtx)
 					if err != nil && err != ctx.Err() {
 						cancel()
 					}
-					mq.CancelWorkers(pid)
+					ourmq.CancelWorkers(pid)
 					localCancel()
 					wg.Done()
 				}()
```
Comment (lafriks) [inline modules/queue/manager.go]:
```suggestion
				}(mq)
```

File: modules/queue/manager.go [hunk 3]
```diff
@@ -183,14 +183,15 @@ func (m *Manager) FlushAll(baseCtx context.Context, timeout time.Duration) error
 			}
 			allEmpty = false
 			if flushable, ok := mq.Managed.(Flushable); ok {
+				ourmq := mq
```
Comment (lafriks) [inline modules/queue/manager.go]:
```suggestion
```

File: modules/queue/manager.go [hunk 4]
```diff
@@ -183,14 +183,15 @@ func (m *Manager) FlushAll(baseCtx context.Context, timeout time.Duration) error
...
 				go func() {
 					localCtx, localCancel := context.WithCancel(ctx)
-					pid := mq.RegisterWorkers(1, start, hasTimeout, end, localCancel, true)
+					pid := ourmq.RegisterWorkers(1, start, hasTimeout, end, localCancel, true)
```
Comment (lafriks) [inline modules/queue/manager.go]:
```suggestion
					pid := q.RegisterWorkers(1, start, hasTimeout, end, localCancel, true)
```

File: modules/queue/manager.go [hunk 5]
```diff
@@ -183,14 +183,15 @@ func (m *Manager) FlushAll(baseCtx context.Context, timeout time.Duration) error
...
 					err := flushable.FlushWithContext(localCtx)
 					if err != nil && err != ctx.Err() {
 						cancel()
 					}
-					mq.CancelWorkers(pid)
+					ourmq.CancelWorkers(pid)
```
Comment (lafriks) [inline modules/queue/manager.go]:
```suggestion
					q.CancelWorkers(pid)
```

**Review by lafriks — CHANGES_REQUESTED** [review]
This would be more correct imho

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Comment by zeripath** [comment]
Make lg-tm work

**Pull request merged**